app/code/Magento/InventoryReservations/Model/ResourceModel/CleanupReservations.php

```diff
@@ -51,7 +51,7 @@ public function execute(): void
                 $reservationTable,
                 ['GROUP_CONCAT(' . ReservationInterface::RESERVATION_ID . ')']
             )
-            ->group([ReservationInterface::STOCK_ID, ReservationInterface::SKU])
+            ->group("JSON_EXTRACT(metadata, '$.object_id')")
```

Comment:
It's really unfortunate, but Magento still [supports](https://devdocs.magento.com/guides/v2.3/install-gde/system-requirements-tech.html#database) MySQL versions as old as 5.6, which do not have any of the JSON functions. 

We would need to come up with a more versatile solution here to get it merged in the core